datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/concurrency/GraphQLWorkerPoolThreadFactory.java

```diff
@@ -0,0 +1,25 @@
+package com.linkedin.datahub.graphql.concurrency;
+
+import java.util.concurrent.ThreadFactory;
+
+public class GraphQLWorkerPoolThreadFactory implements ThreadFactory {
+
+  private static int threadInitNumber;
+
+  private static synchronized int nextThreadNum() {
+    return threadInitNumber++;
+  }
+
+  private long stackSize;
+
+  public GraphQLWorkerPoolThreadFactory(long stackSize) {
+    this.stackSize = stackSize;
+  }
+
+  @Override
+  public final Thread newThread(Runnable runnable) {
+    Thread thread = new Thread(null, runnable, "GraphQLWorkerThread-" + nextThreadNum(), stackSize);
```

Comment:
Should we create/pass a thread group? Feels like it could be useful - at least for debugging.

Reply:
Would this help more than the naming? I didn't see any of the base ThreadFactories specifying a thread group so wasn't sure how useful it is.

Reply:
I was thinking mainly for debugging/naming, it provides a variable that has the list of all GraphQL threads. Maybe in the future it could be useful for having dedicated thread groups for different tasks.

Reply:
Yeah that's why I gave the custom naming scheme, but I can add a group too if we think it's useful.